src/discord-cluster-manager/cogs/leaderboard_cog.py

```diff
@@ -251,76 +273,132 @@ async def leaderboard_create(
             try:
                 date_value = datetime.strptime(deadline, "%Y-%m-%d %H:%M")
             except ValueError:
-                # If that fails, try parsing just the date (will set time to 00:00)
-                date_value = datetime.strptime(deadline, "%Y-%m-%d")
+                try:
+                    date_value = datetime.strptime(deadline, "%Y-%m-%d")
+                except ValueError:
+                    raise ValueError(
```

Comment:
I think we should do `interaction.response.send_message("Invalid date format. Please use...")` and then return here. It's a simpler control flow.

Reply:
Updated this, there was also a mistake in here regarding bad implementations I forgot to delete. It's been simplified.